File: module_wanandroid/src/main/java/com/oreooo/wanandroid/base/AbstractPresenter.java
package com.oreooo.wanandroid.base;

import io.reactivex.disposables.Disposable;

/**
 *  Presenter 基类接口
 * @author oreo
 * @date 2020/05/09
 */
public interface AbstractPresenter<T extends AbstractView> {

    void attachView(T view);

    void detachView();

    void addSubscribe(Disposable disposable);
}

File: lib_baselibrary/src/main/java/com/oreooo/baselibrary/newmvp/AbstractPresenter.kt
package com.oreooo.baselibrary.newmvp

import io.reactivex.disposables.Disposable

/**
 *  Presenter 基类接口
 * @author oreo
 * @date 2020/05/20
 */
interface AbstractPresenter<T: AbstractView> {
    fun attachView(view: T)

    fun detachView()

    fun addSubscribe(disposable: Disposable?)
}